core/src/main/java/com/datastrato/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -102,4 +110,65 @@ public static void checkRoleNamespace(Namespace namespace) {
         "Role namespace must have 3 levels, the input namespace is %s",
         namespace);
   }
+
+  public static void checkPermission(String metalake, SecurableObject object) {
+
+    if (!satisfyPrivileges(metalake, object)) {
+      throw new ForbiddenException(
+          "Securable object %s doesn't have the one of privileges %s",
+          object,
+          StringUtils.join(
+              object.privileges().stream().map(Privilege::name).collect(Collectors.toList()), ","));
+    }
+  }
+
+  public static boolean satisfyPrivileges(String metalake, SecurableObject object) {
+    if (object.privileges().isEmpty()) {
+      return true;
+    }
+
+    String currentUser = PrincipalUtils.getCurrentUserName();
+
+    AccessControlManager accessControlManager = GravitinoEnv.getInstance().accessControlManager();
+    List<RoleEntity> roles;
+    try {
+      roles = accessControlManager.getRolesByUserFromMetalake(metalake, currentUser);
+
+      for (RoleEntity role : roles) {
+        for (Privilege privilege : object.privileges()) {
+          // The deny privilege is prior to the allow privilege. It means that one entity has the
+          // deny privilege and allow privilege at the same time. The entity doesn't have the
+          // privilege.
+          if (role.hasPrivilegeWithCondition(object, privilege.name(), Privilege.Condition.DENY)) {
+            continue;
+          }
+
+          if (role.hasPrivilegeWithCondition(object, privilege.name(), Privilege.Condition.ALLOW)) {
```

Comment:
Considering the following case: 
Privileges of `object` is:  Create table: allow, create database: allow
Privileges of current user: create table: allow, create table: deny

So the final result is true????

Reply:
Yes.